proto/workflow_execution_service.proto

```diff
@@ -0,0 +1,237 @@
+
+
+// The Workflow Execution Service presents an interoperable layer for executing workflows
+// using an HTTP interface. By passing WES messages to a service connected to some execution
+// engine one can carry out a computational task.
+service WorkflowExecutionService {
+
+  // Get information about Workflow Execution Service.  May include information related (but
+  // not limited to) the workflow descriptor formats, versions supported, the WES API versions supported, and information about general the service availability.
+  rpc GetServiceInfo(service_info_request) returns (service_info) {
+    option (google.api.http) = {
+      get: "/ga4gh/wes/v1/service-info"
+    };
+  }
+
+  // Run a workflow, this endpoint will allow you to create a new workflow request and
+  // retrieve its tracking ID to monitor its progress.  An important assumption in this
+  // endpoint is that the workflow_params JSON will include parameterizations along with
+  // input and output files.  The latter two may be on S3, Google object storage, local filesystems,
+  // etc.  This specification makes no distinction.  However, it is assumed that the submitter
+  // is using URLs that this system both understands and can access. For Amazon S3, this could
+  // be accomplished by given the credentials associated with a WES service access to a
+  // particular bucket.  The details are important for a production system and user on-boarding
+  // but outside the scope of this spec.
+  rpc RunWorkflow(workflow_request) returns (workflow_run_id) {
+    option (google.api.http) = {
+      post: "/ga4gh/wes/v1/workflows"
+      body: "*"
+    };
+  }
+
+  // Get quick status info about a running workflow.
+  rpc GetWorkflowStatus(workflow_run_id) returns (workflow_status) {
+      option (google.api.http) = {
+        get: "/ga4gh/wes/v1/workflows/{workflow_id}/status"
+      };
+  }
+
+  // Get detailed info about a running workflow.
+  rpc GetWorkflowLog(workflow_run_id) returns (workflow_log) {
+      option (google.api.http) = {
+        get: "/ga4gh/wes/v1/workflows/{workflow_id}"
+      };
+  }
+
+  // Cancel a running workflow.
+  rpc CancelJob(workflow_run_id) returns (workflow_run_id) {
+    option (google.api.http) = {
+      delete: "/ga4gh/wes/v1/workflows/{workflow_id}"
+    };
+  }
+
+  // List the workflows, this endpoint will list the workflows in order of oldest to newest.
+  // There is no guarantee of live updates as the user traverses the pages, the behavior should be
+  // decided (and documented) by each implementation.
+  // To monitor a given execution, use GetWorkflowStatus or GetWorkflowLog.
+  rpc ListWorkflows(workflow_list_request) returns (workflow_list_response) {
+    option (google.api.http) = {
+      get: "/ga4gh/wes/v1/workflows"
+    };
+  }
+
+}
+
+// To execute a workflow, send a workflow request including all the details needed to begin downloading
+// and executing a given workflow.
+message workflow_request {
```

Comment:
Protobuf style guide suggests camel case for message names ( https://developers.google.com/protocol-buffers/docs/style )